crates/tako/src/worker/rpc.rs

```diff
@@ -457,6 +443,47 @@ pub async fn connection_initiator(
     }
 }
 
+fn send_overview_to_server(
+    state: RefMut<'_, WorkerState>,
+    overview_request: OverviewRequest,
+) {
+    let message = FromWorkerMessage::Overview(WorkerOverview {
+        id: state.worker_id,
+        running_tasks: state
+            .running_tasks
+            .iter()
+            .map(|tr| {
+                let task = tr.get();
+                let allocation: ResourceAllocation = task.resource_allocation().unwrap().clone();
+                (task.id, allocation)
+                // TODO: Modify this when more cpus are allowed
+            })
+            .collect(),
+        placed_data: state.data_objects.keys().copied().collect(),
+
+        hw_state: match overview_request.enable_hw_overview {
+            true => Some(WorkerHwStateMessage {
+                state: state.hardware_state.clone(),
+            }),
+            false => None,
+        },
+    });
+    state.send_message_to_server(message);
+}
+
+async fn update_overview(state: WorkerStateRef, overview_update_interval: Duration) {
+    let mut update_interval = tokio::time::interval(overview_update_interval);
+    loop {
+        update_interval.tick().await;
+        send_overview_to_server(
+            state.get_mut(),
+            OverviewRequest {
```

Comment:
I'll fix these in a later commit, this commit is just to ask the question about the futures::pending confusion